test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+        # on the hook script, where MS defines 1 value, while the 2 sister
+        # Moms define the same value. For 'string' type, it will be a
+        # union of all values obtained from sister moms and local mom, and
+        # the result will be in JSON-format.
+        #
+        # foo_str is for testing normal values.
+        # foo_str2 is for testing non-JSON format value received from MS.
+        # foo_str3 is for testing non-JSON format value received from a sister
+        # mom.
+        # foo_str4 is for testing MS-only set values.
+        #
+        # For string_array type  resource 'stra', it is not accumulated but
+        # will be set to last seen value from a mom epilogue hook.
+        foo_str3_val = '{"a":6,"b":"some value #$%^&*@","c":54.4,"d":"32.5gb"}'
+        self.server.expect(JOB, {
+            'job_state': 'F',
+            'resources_used.foo_f': '0.09',
+            'resources_used.foo_i': '9',
+            'resources_used.foo_str': '{"seven":7}',
+            'resources_used.foo_str2': 'seven',
+            'resources_used.foo_str3': foo_str3_val,
+            'resources_used.stra': "\"glad,elated\",\"happy\"",
+            'resources_used.vmem': '9gb',
+            'resources_used.cput': '00:00:10',
+            'resources_used.ncpus': '1'},
+            extend='x', offset=10, id=jid)
+
+        # Match accounting_logs entry
+
+        acctlog_match = 'resources_used.foo_f=0.09'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.foo_i=9'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.foo_str={"seven":7}'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.vmem=9gb'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.cput=00:00:10'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.foo_str2=seven'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.ncpus=1'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.foo_str3='
+        acctlog_match += '\{"a"\:6\,"b"\:"some value \#\$\%\^\&\*\@"\,'
+        acctlog_match += '"c"\:54\.4\,"d"\:"32\.5gb"\}'
```

Comment:
I could but then I run into the "too long" of a line in pep8. This is the reason why I had to break up the variable assignment this way.